mobile/lib/pages/article_selection_page.dart

```diff
@@ -0,0 +1,142 @@
+import 'dart:async';
+import 'dart:convert';
+
+import 'package:flutter/material.dart';
+import 'package:flutter_widget_from_html_core/flutter_widget_from_html_core.dart';
+import 'package:http/http.dart' as http;
+import 'package:radioaktywne/components/utility/color_shadowed_card_2.dart';
+import 'package:radioaktywne/components/utility/ra_progress_indicator.dart';
+import 'package:radioaktywne/components/utility/refreshable_fetch_widget.dart';
+import 'package:radioaktywne/extensions/extensions.dart';
+import 'package:radioaktywne/models/article_info.dart';
+import 'package:radioaktywne/pages/article_page.dart';
+
+class ArticleSelectionPage extends StatelessWidget {
+  const ArticleSelectionPage({
+    super.key,
+    this.timeout = const Duration(seconds: 15),
+  });
+
+  final Duration timeout;
+
+  static const EdgeInsets _pagePadding = EdgeInsets.symmetric(horizontal: 26);
+
+  // Single URL that returns all articles
+  // TODO: Lazy loading and loading every article not one page of articles
+  static final Uri _infoUrl = Uri.parse('https://radioaktywne.pl/wp-json/wp/v2/posts?_embed=true&page=1&per_page=16');
+
+  Future<List<ArticleInfo>> _fetchArticles() async {
+    final articles = <ArticleInfo>[];
+
+    try {
+      final response = await http.get(_infoUrl).timeout(timeout);
+      final jsonDataList = jsonDecode(response.body) as List<dynamic>;
+
+      for (final jsonData in jsonDataList) {
+        final article = ArticleInfo.fromJson(jsonData as Map<String, dynamic>);
+        articles.add(article);
```

Comment:
With a little twist, but you are right it does exactly that. Therfore I modifed the code to this:
```Dart
Future<Iterable<ArticleInfo>> _fetchArticles() async {
  try {
    return await fetchData(_infoUrl, ArticleInfo.fromJson);
  } on TimeoutException catch (_) {
    return [];
  }
}
```

Also since it is now `Iterable` type I needed to change this line of code because you can't access the elements of `Iterable` type using `[]`

```Dart
itemBuilder: (context, index) {
    final article = articles.elementAt(index);
```